Automattic/jetpack
# Jetpack: Fix Backup item not displaying when site has a plan but Backup plugin is not active

The `Backup` submenu item needs to be visible (and pointing to Jetpack Cloud) when the site has a Jetpack VaultPress Backup plan and the standalone plugin is not active.

We used to validate if the `Jetpack_Backup` class exist to determine whether the plugin is active, and given that we moved that class to the backup package in #23531, it will always respond that it is active, so will hide the submenu item.

## Proposed changes:

* Validate if the plugin is active using [My Jetpack](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/my-jetpack/src/products/class-backup.php#L211) instead of the existence of the `Jetpack_Backup` class, which is now available as part of the Backup package.
* Rename `Backup` submenu to `VaultPress`
* If the site has a Backup and a Scan plan, let's separate the submenu instead of using `Backup & Scan`. This will properly let us handle the new branding.

## Screenshots:
| Scenario | Before | After |
|---|---|---|
| Backup plan active, Backup SA plugin inactive | ![image](https://user-images.githubusercontent.com/1488641/216125968-cd1a9511-07df-4ad5-ba81-484351cec736.png) | ![image](https://user-images.githubusercontent.com/1488641/216125878-0675c400-3a49-42ce-9507-d33785bc58d6.png) |
| Backup and scan plan active, Backup plugin and Protect inactive | ![image](https://user-images.githubusercontent.com/1488641/216129751-d1a241f8-8465-42bf-98a0-d22fccbd8132.png) | ![image](https://user-images.githubusercontent.com/1488641/216129305-8b947ca1-6842-43db-bfe8-de0aae48b6e7.png) |

### Other information:


## Jetpack product discussion


* pbNhbs-5aa-p2
* p1675100395340909-slack-CS8UYNPEE

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






### Testing VaultPress submenu
* Spin up a Jurassic Ninja site with Jetpack.
* Connect the site to Jetpack.
* Purchase a Jetpack VaultPress Backup (10GB) plan.
* Ensure the backup plan is active and the backup plugin is installed and active.
* Navigate to Jetpack > My Jetpack
* Validate the `Backup` submenu item pointing to Jetpack Cloud is not visible.
* Deactivate the backup standalone plugin.
* Validate the `Backup` submenu item is available on the menu and points to Jetpack Cloud.

### Testing Scan submenu
* Spin up a Jurassic Ninja site with Jetpack.
* Connect the site to Jetpack.
* Purchase a Jetpack Scan Daily plan.
* Ensure the scan plan is active and the Protect plugin is installed and active.
* Navigate to Jetpack > My Jetpack
* Validate the `Scan` submenu item pointing to Jetpack Cloud is not visible.
* Deactivate the Protect standalone plugin.
* Validate the `Scan` submenu item is available on the menu and points to Jetpack Cloud.

### Testing VaultPress and Scan submenus
* Spin up a Jurassic Ninja site with Jetpack.
* Connect the site to Jetpack.
* Purchase a Jetpack Security plan (or a Backup plan and Scan plan independently)
* Ensure the backup and scan plan is active and Backup / Protect plugins are installed and active.
* Navigate to Jetpack > My Jetpack
* Validate the `VaultPress` and `Scan` submenu items pointing to Jetpack Cloud are not visible.
* Deactivate the Protect plugin.
* Validate now that `Scan` submenu item is available on the menu and points to Jetpack Cloud.
* Deactivate the Backup standalone plugin.
* Validate now taht `VaultPress` submenu item is now available on the menu and points to Jetpack Cloud.
